src/mock/class_addressGenerator.py

```diff
@@ -0,0 +1,109 @@
+        'privencryptionkey': '5JaeFJs8iPcQT3N8676r3gHKvJ5mTWXy1VLhGCEDqRs4vpvpxV8'
+    },
+    'BM-2cUd2dm8MVMokruMTcGhhteTpyRZCAMhnA': {
+        'privsigningkey': '5JnJ79nkcwjo4Aj7iG8sFMkzYoQqWfpUjTcitTuFJZ1YKHZz98J',
+        'privencryptionkey': '5JXgNzTRouFLqSRFJvuHMDHCYPBvTeMPBiHt4Jeb6smNjhUNTYq'
+    },
+    'BM-2cWyvL54WytfALrJHZqbsDHca5QkrtByAW': {
+        'privsigningkey': '5KVE4gLmcfYVicLdgyD4GmnbBTFSnY7Yj2UCuytQqgBBsfwDhpi',
+        'privencryptionkey': '5JTw48CGm5CP8fyJUJQMq8HQANQMHDHp2ETUe1dgm6EFpT1egD7'
+    },
+    'BM-2cTE65PK9Y4AQEkCZbazV86pcQACocnRXd': {
+        'privsigningkey': '5KCuyReHx9MB4m5hhEyCWcLEXqc8rxhD1T2VWk8CicPFc8B6LaZ',
+        'privencryptionkey': '5KBRpwXdX3n2tP7f583SbFgfzgs6Jemx7qfYqhdH7B1Vhe2jqY6'
+    },
+    'BM-2cX5z1EgmJ87f2oKAwXdv4VQtEVwr2V3BG': {
+        'privsigningkey': '5K5UK7qED7F1uWCVsehudQrszLyMZxFVnP6vN2VDQAjtn5qnyRK',
+        'privencryptionkey': '5J5coocoJBX6hy5DFTWKtyEgPmADpSwfQTazMpU7QPeART6oMAu'
+    }
+}
+
+
+UISignalQueue = Queue.Queue()
+addressGeneratorQueue = Queue.Queue()
+
+
+class StoppableThread(threading.Thread):
+    """Base class for application threads with stopThread method"""
+    name = None
+    logger = logging.getLogger('default')
+
+    def __init__(self, name=None):
+        if name:
+            self.name = name
+        super(StoppableThread, self).__init__(name=self.name)
+        self.stop = threading.Event()
+        self._stopped = False
+        random.seed()
+        self.logger.info('Init thread %s', self.name)
+
+    def stopThread(self):
+        """Stop the thread"""
+        self._stopped = True
+        self.stop.set()
+
+
+class FakeAddressGenerator(StoppableThread):
+    """A thread for creating fake addresses"""
+    name = "addressGenerator"
+    address_list = list(fake_addresses.keys())
+
+    def stopThread(self):
+        try:
+            addressGeneratorQueue.put(("stopThread", "data"))
+        except:
+            pass
+        super(FakeAddressGenerator, self).stopThread()
+
+    def run(self):
+        """
+        Process the requests for addresses generation
+        from `.queues.addressGeneratorQueue`
+        """
+        while state.shutdown == 0:
+            queueValue = addressGeneratorQueue.get()
+            streamNumber = 1
+            try:
+                address = self.address_list.pop(0)
+                label = queueValue[3]
+
+                BMConfigParser().add_section(address)
+                BMConfigParser().set(address, 'label', label)
+                BMConfigParser().set(address, 'enabled', 'true')
+                BMConfigParser().set(
+                    address, 'privsigningkey', fake_addresses[address]['privsigningkey'])
+                BMConfigParser().set(
+                    address, 'privencryptionkey', fake_addresses[address]['privencryptionkey'])
+                BMConfigParser().save()
+
+                UISignalQueue.put((
```

Comment:
We can probably skip using `UISignalQueue` in this mock class for the time being. Maybe it will be needed later but at the moment it isn't.

Reply:
Ok, For now, I am removing it but it will be required in the test_create_random_address.py test case because here I am checking  created address in spinner or my address list so this will be required in that case

Reply:
Ok.